controllers/humiocluster_controller.go

```diff
@@ -537,34 +555,46 @@ func (r *HumioClusterReconciler) ensureViewGroupPermissionsConfigMap(ctx context
 	if viewGroupPermissionsConfigMapData == "" {
 		viewGroupPermissionsConfigMap, err := kubernetes.GetConfigMap(ctx, r, ViewGroupPermissionsConfigMapName(hc), hc.Namespace)
 		if err == nil {
-			if err = r.Delete(ctx, viewGroupPermissionsConfigMap); err != nil {
-				r.Log.Error(err, "unable to delete view group permissions config map")
+			// TODO: refactor and move deletion to cleanupUnusedResources
```

Comment:
Same for this. Also, maybe we could extract the functionality of this if > if into a method as it's duplicate to the `extraKafkaConfigsConfigMapData` one.

Actually, a lot of the `ensureViewGroupPermissionsConfigMap` method logic is similar to the one for KafkaConfigMap ^^

Reply:
True, they all follow the same logic for creating/updating/deleting configmaps, but the PR is meant to fix the reported bugs and not so much reduce the code duplication, since we can do that in a followup. That followup could also move the cleanup logic to `cleanupUnusedResources` function as indicated in the comment.